wrapper/src/main/java/software/amazon/jdbc/util/UpdateTracker.java

```diff
@@ -0,0 +1,37 @@
+package software.amazon.jdbc.util;
+
+import java.sql.Connection;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class UpdateTracker {
+  private static final long FIVE_MINUTES_MS = 5 * 60 * 1000;
+  private final ConcurrentHashMap<Integer, Long> connectionUpdates;
+
+  public UpdateTracker() {
+      this.connectionUpdates = new ConcurrentHashMap<>();
+  }
+
+  public void recordUpdate(Connection connection) {
+    if (connection == null) return;
+    int connectionId = System.identityHashCode(connection);
+    long curTime = System.currentTimeMillis();
```

Comment:
`System.currentTimeMillis()` is relatively fast, however it doesn't offer sub-millisecond level granularity. Maybe use `System.nanoTime()` instead?